services/app-api/src/resolvers/contractResolvers/updateContractResolver.ts

```diff
@@ -0,0 +1,110 @@
+import { ForbiddenError, UserInputError } from 'apollo-server-lambda'
+import {
+    isCMSUser,
+    convertContractWithRatesToUnlockedHPP,
+} from '../../domain-models'
+import type { MutationResolvers } from '../../gen/gqlServer'
+import { logError } from '../../logger'
+import type { Store } from '../../postgres'
+import {
+    setResolverDetailsOnActiveSpan,
+    setErrorAttributesOnActiveSpan,
+} from '../attributeHelper'
+import type { LDService } from '../../launchDarkly/launchDarkly'
+import { GraphQLError } from 'graphql'
+
+export function updateContractResolver(
+    store: Store,
+    launchDarkly: LDService
+): MutationResolvers['updateContract'] {
+    return async (_parent, { input }, context) => {
+        const { user, span } = context
+        setResolverDetailsOnActiveSpan('updateContract', user, span)
+        const ratesDatabaseRefactor = await launchDarkly.getFeatureFlag(
+            context,
+            'rates-db-refactor'
+        )
+
+        if (ratesDatabaseRefactor) {
+            // This resolver is only callable by CMS users
+            if (!isCMSUser(user)) {
+                logError(
+                    'updateContract',
+                    'user not authorized to update contract'
+                )
+                setErrorAttributesOnActiveSpan(
+                    'user not authorized to update contract',
+                    span
+                )
+                throw new ForbiddenError(
+                    'user not authorized to update contract'
+                )
+            }
+
+            const contract = await store.findContractWithHistory(input.id)
+            if (contract instanceof Error) {
+                throw contract
+            }
+
+            const hasSubmitInfo =
```

Comment:
It was called out to check that a revision has submitInfo but I was wondering if a status check for `SUBMITTED` would work instead?